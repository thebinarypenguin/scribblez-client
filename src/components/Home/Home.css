#Home .Hero {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 3rem;
}

#Home p {
  padding: 0 1.5rem;
  line-height: 1.5;
  font-size: 1rem;
  padding: 3rem 2rem 4rem 2rem;
}
