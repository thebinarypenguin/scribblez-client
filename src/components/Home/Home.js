import React from 'react';
import UserContext from '../../contexts/UserContext';
import TopBar from '../TopBar/TopBar';
import Utils from '../../services/Utils';

import './Home.css';

class Home extends React.Component {

  static contextType = UserContext;

  constructor(props) {
    super(props);

    this.generateLinks = this.generateLinks.bind(this);
  }

  generateLinks() {

    let links = [];

    if (Utils.isTokenActive(window.localStorage.getItem('token'))) {

      links = [
        { name: 'Feed',     target: '/feed'     },
        { name: 'Notes',    target: '/notes'    },
        { name: 'Sign Out', target: '/sign-out' },
      ];

    } else {

      links = [
        { name: 'Feed',    target: '/feed'    },
        { name: 'Sign Up', target: '/sign-up' },
        { name: 'Sign In', target: '/sign-in' },
      ];
    }

    return links;
  }

  render() {

    return (

      <div id="site-container">

      <div id="home-container">

        <TopBar links={this.generateLinks()} />

        <div id="hero">
          <p>Hero Content</p>
        </div>

        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum
          molestias dolor rem accusantium consectetur molestiae illo labore,
          officiis eius, fugit recusandae, id assumenda ipsam. Ratione explicabo
          facilis vel praesentium cumque. Doloribus sit ipsam commodi doloremque
          quisquam repellendus, tempora reiciendis fuga rem aspernatur, voluptatem
          nisi explicabo delectus dolore dicta omnis odio aliquam culpa.
        </p>

      </div>

      </div>
    );
  }
}

export default Home;
