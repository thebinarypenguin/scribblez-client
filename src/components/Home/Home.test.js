import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Home from './Home';

it('renders without crashing', () => {

  const container = document.createElement('div');

  const subject = (
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  ReactDOM.render(subject, container);
  ReactDOM.unmountComponentAtNode(container);
});

it('renders the UI as expected', () => {

  const subject = (
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  const tree = renderer.create(subject).toJSON();
  expect(tree).toMatchSnapshot();
});
