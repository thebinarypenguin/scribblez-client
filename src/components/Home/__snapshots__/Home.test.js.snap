// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="Home"
>
  <div
    id="TopBar"
  >
    <div
      className="Container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          Scribblez
        </h1>
      </a>
      <ul>
        <li>
          <a
            href="/feed"
            onClick={[Function]}
          >
            Feed
          </a>
        </li>
        <li>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/sign-in"
            onClick={[Function]}
          >
            Sign In
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Container"
  >
    <p>
      Scribblez is a note-sharing service that allows you to share your ideas with the world. Don't like sharing? That's okay you can make your notes private. We won't tell anyone.
    </p>
  </div>
</div>
`;
